client/errors.go

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2023 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package client
+
+// Error is the real value of response.Result when an error type response occurs.
+type Error struct {
+	Message string      `json:"message"`
+	Kind    ErrorKind   `json:"kind,omitempty"`
+	Value   interface{} `json:"value,omitempty"`
+}
+
+func (e *Error) Error() string {
+	return e.Message
+}
+
+type ErrorKind string
+
+// Error kinds for use as a response result.
+const (
+	ErrorKindLoginRequired     = ErrorKind("login-required")
+	ErrorKindNoDefaultServices = ErrorKind("no-default-services")
```

Comment:
This name doesn't really tell us what's wrong. Was there an expectation of default services existing?

Reply:
Here "default services" is being used in the context of an operation. When we ask for an autostart, this will start whatever was supposed to start "by default", unless there's nothing to start by default. That's what the error is saying.

It's unfortunately too late to bikeshed about the error kind, because those are public constants exposed in the API, and real code depends on them.